File: src/main/js/models/nodes/UnaryPlusOperationNode.ts
import { Node } from "./Node";
import { NodeType } from "./NodeType";

/**
 * Unary plus operation.
 */
export class UnaryPlusOperationNode extends Object implements Node {
  readonly type: NodeType = NodeType.UnaryPlusOperation;
  readonly operand: Node;

  constructor(operand: Node) {
    super();
    this.operand = operand;
  }

  toString() {
    return `+${this.operand.toString()}`;
  }
}

File: src/main/java/io/protobase/f7/models/UnaryPlusOperationNode.java
package io.protobase.f7.models;

import com.google.common.base.MoreObjects;

/**
 * Unary plus operation.
 */
public class UnaryPlusOperationNode extends BaseObject implements Node {
  private Node operand;

  /**
   * Create new node with an operand.
   *
   * @param operand value
   */
  public UnaryPlusOperationNode(Node operand) {
    this.operand = operand;
  }

  /**
   * Get the operand node.
   *
   * @return node.
   */
  public Node getOperand() {
    return operand;
  }

  @Override
  public String toString() {
    return MoreObjects.toStringHelper(this)
        .add("operand", operand)
        .toString();
  }

  @Override
  public Object[] significantAttributes() {
    return new Object[]{
        operand
    };
  }
}
